stress: failed test in cockroach/util/retry/retry.test: TestRetryExceedsMaxBackoff

Binary: cockroach/static-tests.tar.gz sha: https://github.com/cockroachdb/cockroach/commits/b1125fa08614aa770703b67a18dfb9856e2cd4ad

Stress build found a failed test:

```
=== RUN   TestRetryExceedsMaxBackoff
--- FAIL: TestRetryExceedsMaxBackoff (1.00s)
	retry_test.go:44: expected retry loop to run for 20ms, actually ran for 1.00013201s
```

Run Details:

```
6485 runs so far, 0 failures, over 5s
12793 runs so far, 0 failures, over 10s
18870 runs so far, 0 failures, over 15s
24458 runs so far, 0 failures, over 20s
29262 runs so far, 0 failures, over 25s
34010 runs so far, 0 failures, over 30s
38666 runs so far, 0 failures, over 35s
43233 runs so far, 0 failures, over 40s
47710 runs so far, 0 failures, over 45s
52148 runs so far, 0 failures, over 50s
56451 runs so far, 0 failures, over 55s
60675 runs so far, 0 failures, over 1m0s
64817 runs so far, 0 failures, over 1m5s
68960 runs so far, 0 failures, over 1m10s
72934 runs so far, 0 failures, over 1m15s
76099 runs so far, 0 failures, over 1m20s
79977 runs so far, 0 failures, over 1m25s
83575 runs so far, 0 failures, over 1m31s
83576 runs completed, 1 failures, over 1m31s
FAIL
```
Please assign, take a look and update the issue accordingly.

--- a/util/retry/retry_test.go
+++ b/util/retry/retry_test.go
@@ -19,35 +19,24 @@ package retry
 import (
 	"testing"
 	"time"
-
-	"github.com/cockroachdb/cockroach/util/timeutil"
 )
 
 func TestRetryExceedsMaxBackoff(t *testing.T) {
 	opts := Options{
 		InitialBackoff: time.Microsecond * 10,
-		MaxBackoff:     time.Microsecond * 10,
-		Multiplier:     1e5,
-		MaxRetries:     1,
+		MaxBackoff:     time.Microsecond * 100,
+		Multiplier:     2,
+		MaxRetries:     10,
 	}
 
-	// tolerate a duration two orders of magnitude lower than if MaxBackoff is
-	// not respected.
-	fudgeFactor := opts.Multiplier / 100
-
-	attempts := 0
-	start := timeutil.Now()
-	for r := Start(opts); r.Next(); attempts++ {
-	}
-
-	duration := time.Since(start)
-
-	if expDuration := time.Duration(fudgeFactor) * time.Duration(opts.MaxRetries+1) * opts.MaxBackoff; duration > expDuration {
-		t.Errorf("expected retry loop to run for %s, actually ran for %s", expDuration, duration)
-	}
-
-	if expAttempts := opts.MaxRetries + 1; attempts != expAttempts {
-		t.Errorf("expected %d attempts, got %d attempts", expAttempts, attempts)
+	r := Start(opts)
+	r.opts.RandomizationFactor = 0
+	for i := 0; i < 10; i++ {
+		d := r.retryIn()
+		if d > opts.MaxBackoff {
+			t.Fatalf("expected backoff less than max-backoff: %s vs %s", d, opts.MaxBackoff)
+		}
+		r.currentAttempt++
 	}
 }
 

